packages/cozy-harvest-lib/src/components/hoc/withDomain.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { PureComponent } from 'react'
+import memoize from 'lodash/memoize'
+import PropTypes from 'prop-types'
+
+const getGlobalDomain = memoize(() => {
+  try {
+    return document.querySelector('[role=application]').dataset.cozyDomain
+  } catch (error) {
+    throw new Error(
+      'Cannot retrieve data-cozy-domain attribute on [role=application] element'
+    )
+  }
+})
+
+export const withDomain = WrappedComponent => {
+  class ComponentWithDomain extends PureComponent {
+    render() {
+      const contextDomain = this.context.domain
+      const domain = contextDomain || getGlobalDomain()
+      return <WrappedComponent {...this.props} domain={domain} />
+    }
+  }
+
+  ComponentWithDomain.contextTypes = {
+    domain: PropTypes.string.isRequired
+  }
+
+  ComponentWithDomain.displayName = `withDomain(${WrappedComponent.name ||
+    WrappedComponent.displayName})`
+
+  return ComponentWithDomain
+}
```

Comment:
I think instead of using this, we should use withClient and client.stackClient.uri .